src/main/cc/wfa/virtual_people/core/model/utils/distributed_consistent_hashing.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 The Cross-Media Measurement Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "wfa/virtual_people/core/model/utils/distributed_consistent_hashing.h"
+
+#include <cmath>
+
+#include "absl/memory/memory.h"
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+#include "src/farmhash.h"
+
+namespace wfa_virtual_people {
+
+absl::StatusOr<std::unique_ptr<DistributedConsistentHashing>>
+DistributedConsistentHashing::Build(
+    std::unique_ptr<std::vector<DistributionChoice>> distribution) {
```

Comment:
It's not clear to me why this needs to be a unique_ptr. Could you clarify? Vectors are efficiently moveable -- the unique_ptr seems like an unnecessary indirection. It also adds a bug: what if it's nullptr? Then the `distribution->empty()` check below will crash.

Reply:
Done.